### DISCUSSION TITLE
Cannot promote queue when home node is down

### DISCUSSION THREAD

**Discussion by polaris-alioth** [rabbitmq/rabbitmq-server#3590] [Other]
I have a mirrored queue named testeat, the master node is rabbitmq-1 and slave node is rabbitmq-0.
![image](https://user-images.githubusercontent.com/92571231/137826960-a891746a-e4e0-4e44-bd5f-9a885f756e1f.png)
rabbitmq-0 and rabbitmq-1 instances is down successively
When I started the rabbitmq-1. The queue is in down status. 
![image](https://user-images.githubusercontent.com/92571231/137827299-c24dd249-c413-4ca4-8c02-b3fe2addb02a.png)
Cannot promote the queue when home node is down.

**Comment by michaelklishin** [comment]
We cannot suggest much with the amount of information provided. See node logs for clues.

On the second screenshot, there are no mirrors for the queues. This can be due to [no mirrors being synchronised](https://www.rabbitmq.com/ha.html#unsynchronised-mirrors) or another reason. Again, there isn't enough information to tell, see node logs for clues. All cases involving no promotable mirrors are logged.

RabbitMQ documentation [explicitly suggests that two node clusters is an unsupported configuration](https://www.rabbitmq.com/clustering.html#node-count) and that classic queue mirroring is deprecated in favor of [quorum queues](https://www.rabbitmq.com/quorum-queues.html) that require an online majority of replicas (and thus 3 or more nodes).

**Reply by polaris-alioth** [reply]
This is my rabbitmq.conf：
_cluster_partition_handling = autoheal_
I set the policy like below, expected that when home node is down, the other node will be promoted to be master.
_rabbitmqctl set_policy -p / ha-allqueue "^" {"ha-mode":"all","ha-sync-mode":"automatic","ha-promote-on-shutdown":"always"}_
Below is the logs:
2021-10-21 17:21:36.041 [info] <0.3228.0> connection <0.3228.0> (172.18.4.97:23049 -> 172.18.1.148:5673): user 'rabbitmq_admin' authenticated and granted access to vhost '/'
2021-10-21 17:21:36.194 [error] <0.3238.0> Channel error on connection <0.3228.0> (172.18.4.97:23049 -> 172.18.1.148:5673, vhost: '/', user: 'rabbitmq_admin'), channel 1:
operation queue.declare caused a channel exception not_found: home node 'rabbit@rabbitmq-1' of durable queue 'test' in vhost '/' is down or inaccessible
